src/Mvc/Mvc.Core/test/Formatters/JsonInputFormatterTestBase.cs

```diff
@@ -110,6 +110,28 @@ public async Task JsonFormatterReadsStringValue()
             Assert.Equal("abcd", stringValue);
         }
 
+        [Fact]
+        public async Task JsonFormatterReadsNonUtf8Content()
+        {
+            // Arrange
+            var content = "☀☁☂☃☄★☆☇☈☉☊☋☌☍☎☏☐☑☒☓☚☛☜☝☞☟☠☡☢☣☤☥☦☧☨☩☪☫☬☭☮☯☰☱☲☳☴☵☶☷☸";
+            var formatter = GetInputFormatter();
+
+            var contentBytes = Encoding.Unicode.GetBytes($"\"{content}\"");
+            var httpContext = GetHttpContext(contentBytes, "applicaiton/json;charset=utf-16");
```

Comment:
`"application/json"` is misspelled. Guess it doesn't matter if the test passes. :)